bcipy/bci_tasks/rsvp/calibration.py

```diff
@@ -102,7 +102,19 @@ def execute(self):
                 if not get_user_input(self.rsvp, self.wait_screen_message,
                                       self.wait_screen_message_color):
                     break
-
+                
+                #Take a break every number of trials defined in parameters.json
+                if not self.parameters['trials_before_break'] == 0:
+                    if (not idx_o == 0) and ((idx_o % self.parameters['trials_before_break']) == 0):
+                        #Update countdown every second
+                        for counter in range(0,self.parameters['break_len']):
+                            self.rsvp.update_task_state(
+                                text=(self.parameters['break_message'] + ' ' + str(self.parameters['break_len'] - counter) + 's'),
```

Comment:
Construct the message outside of the function. Try using f strings (they are very performant).

```
time = self.break_length - counter
message = f'{self.break_message} {time}s'
```